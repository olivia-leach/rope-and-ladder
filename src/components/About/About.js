import React, { Component } from "react";
import { Element } from 'react-scroll'
import './About.scss'

class About extends Component {
  render() {
    return (
      <Element name='about'>
        <div className='fixed-banner about-the-band' />
        <div className='page-section sub-section'>
          <div className="about">
              <div className='about-wrapper'>
                  <div className='section-header'>About the Band</div>
                      <p>Rope and Ladder is the title of the band and rock opera created by Jay Clark, with his writing partner Ryan Dolen.</p>
                      <p>The project has been more than 10 years in the making⁠—in that time, Jay and Ryan pieced together a narrative of epic scope steeped in intrigues of war, love, and betrayal.</p>
                      <p>The album balances delicate piano melodies and classical string voices against distorted guitars and heavy drums, all of which drives the palpable and building tension that accompanies the rising stakes of the plotline.</p>
              </div>
          </div>
        </div>
      </Element>
    );
  }
}

export default About
