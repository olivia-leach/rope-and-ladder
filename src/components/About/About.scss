$break-mobile: 500px;
$break-small: 767px;
$break-medium: 1000px;

.about-wrapper {
    margin: 0 1.3rem;
    line-height: 1.8;
    letter-spacing: 1px;
}

.about {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.fixed-banner {
    position: relative;
    overflow: hidden;
    height: 350px;

    img {
        width: 100%;
    }

    @media screen and (max-width: $break-mobile) {
        height: 150px;
    }
}

.fixed-banner {
    width: 100%;
    height: 40vh;
    background-position: 50%;
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;

    @media screen and (max-width: $break-mobile) {
        height: 220px;
        background-size: 100% auto;
        background-attachment: inherit;
    }

    &.about-the-band {
        background-image: url('../../about_the_band.png')
    }
}