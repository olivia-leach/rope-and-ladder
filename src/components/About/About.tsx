import React from 'react'
import { Element } from 'react-scroll'
import './About.scss'

const About = () => (
  <>
    <div
      className="fixed-banner about-the-band"
      role="img"
      aria-label="band music rock opera rope and ladder"
    />
    <Element className="page-section dark-section" id="about" name="about">
      <div className="about">
        <div className="about-wrapper">
          <div className="section-header">About the Band</div>
          <p>
            Rope and Ladder is the title of the band and rock opera created by
            Jay Clark, with his writing partner Ryan Dolen.
          </p>
          <p>
            The project has been more than 10 years in the making⁠—in that time,
            Jay and Ryan pieced together a narrative of epic scope steeped in
            intrigues of war, love, and betrayal.
          </p>
          <p>
            The album balances delicate piano melodies and classical string
            voices against distorted guitars and heavy drums, all of which
            drives the palpable and building tension that accompanies the rising
            stakes of the plotline.
          </p>
        </div>
      </div>
    </Element>
  </>
)

export default About
