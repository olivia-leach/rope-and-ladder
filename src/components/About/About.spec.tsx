import { render } from '@testing-library/react'
import React from 'react'
import About from './About'

describe('About', () => {
  it('should match snapshot', () => {
    const { container } = render(<About />)
    expect(container).toMatchSnapshot()
  })
})
