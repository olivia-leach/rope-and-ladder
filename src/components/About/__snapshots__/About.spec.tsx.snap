// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match snapshot 1`] = `
<div>
  <div
    aria-label="band music rock opera rope and ladder"
    class="fixed-banner about-the-band"
    role="img"
  />
  <div
    class="page-section dark-section"
    name="about"
  >
    <div
      class="about"
    >
      <div
        class="about-wrapper"
      >
        <div
          class="section-header"
        >
          About the Band
        </div>
        <p>
          Rope and Ladder is the title of the band and rock opera created by Jay Clark, with his writing partner Ryan Dolen.
        </p>
        <p>
          The project has been more than 10 years in the making⁠—in that time, Jay and Ryan pieced together a narrative of epic scope steeped in intrigues of war, love, and betrayal.
        </p>
        <p>
          The album balances delicate piano melodies and classical string voices against distorted guitars and heavy drums, all of which drives the palpable and building tension that accompanies the rising stakes of the plotline.
        </p>
      </div>
    </div>
  </div>
</div>
`;
